Handle passwords with spaces in POP3 server.

evilham opened the pull request:
Closes trac #9100.
https://twistedmatrix.com/trac/ticket/9100

evilham committed:
Handle passwords with spaces in POP3 server.

Closes trac #9100.

evilham committed:
Fix subtle py3 bug in t.m.pop3 detected by higher coverage.

Before this, protocol error messages would send literal "b'error message'" to
the clients.

evilham committed:
Make linter happy in spite of contextual style.

glyph reviewed (approved):
My only question, really, is "what about *two* spaces"?

evilham commented:
Oh, I missed adding such a test, but it will work because we split on `b' '` and then join with that and that trick behaves well without loss of information (read/writelines for files works like that).

```
>>> b'test  2   3'.split(b' ')
[b'test', b'', b'2', b'', b'', b'3']
```

glyph commented:
Thanks for checking on that!
